docs/layer_util.js

```diff
@@ -195,20 +224,25 @@ function addLayerFromId(map, layerName, layerId, index, displayed) {
  * explicitly does not check the actual data. That's why the score layer needs
  * special handling, so deck.gl is forced to re-render it on parameter changes.
  *
- * @param {LayerMapValue} layerMapValue
- * @param {string} layerName
+ * @param {DisplayedLayerData} layerMapValue
+ * @param {number|string} index index of layer. A number unless this is the
+ * score layer
  */
-function addLayerFromFeatures(layerMapValue, layerName) {
-  layerArray[layerMapValue.index] = new deck.GeoJsonLayer({
-    id: layerName,
+function addLayerFromFeatures(layerMapValue, index) {
+  const deckParams = layerMapValue.deckParams;
+  if (!deckParams.colorFunction) {
```

Comment:
When will this trigger?

Reply:
The same time getStyleFunction triggered createStyleFunction before: when this layer is first rendered, and colorFunction hasn't been computed yet.

Reply:
Ah, gotcha, was mixing this up with colorFunctionProperties. So this is an attribute that is only read and accessed in this method?

Reply:
And when we create the score DeckParams object.